Creating a windows inside onClose handler causes Fyne to panic.

Indeed I have encountered that while trying to workaround a missing `WillClose()` handler (needed to prevent closing the window, so I tried to reopen it again). So when I try to create a window within the handler I get:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x45b73d2]

goroutine 1 [running, locked to thread]:
github.com/go-gl/glfw/v3.3/glfw.(*Window).Focus.func1(0x0)
	/Users/ababo/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20200222043503-6f7a984d4dc4/window.go:607 +0x22
github.com/go-gl/glfw/v3.3/glfw.(*Window).Focus(0x0)
	/Users/ababo/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20200222043503-6f7a984d4dc4/window.go:607 +0x2b
fyne.io/fyne/internal/driver/glfw.(*window).create.func3()
	/Users/ababo/go/pkg/mod/fyne.io/fyne@v1.3.0/internal/driver/glfw/window.go:1172 +0x3f5
fyne.io/fyne/internal/driver/glfw.(*gLDriver).runGL(0xc000031b80)
	/Users/ababo/go/pkg/mod/fyne.io/fyne@v1.3.0/internal/driver/glfw/loop.go:96 +0x59b
fyne.io/fyne/internal/driver/glfw.(*gLDriver).Run(0xc000031b80)
	/Users/ababo/go/pkg/mod/fyne.io/fyne@v1.3.0/internal/driver/glfw/driver.go:70 +0x37
fyne.io/fyne/app.(*fyneApp).Run(0xc0000723c0)
	/Users/ababo/go/pkg/mod/fyne.io/fyne@v1.3.0/app/app.go:57 +0x93
main.main()
```
Reproduced on OSX 10.15.5 with Go 1.14.1.

--- a/internal/driver/glfw/loop.go
+++ b/internal/driver/glfw/loop.go
@@ -137,6 +137,10 @@ func (d *gLDriver) runGL() {
 				d.windowLock.Lock()
 				d.windows = newWindows
 				d.windowLock.Unlock()
+
+				if len(newWindows) == 0 {
+					d.Quit()
+				}
 			}
 		}
 	}

--- a/internal/driver/glfw/window.go
+++ b/internal/driver/glfw/window.go
@@ -452,7 +452,7 @@ func (w *window) destroy(d *gLDriver) {
 		w.eventLock.Unlock()
 	}
 
-	if w.master || len(d.windowList()) == 0 {
+	if w.master {
 		d.Quit()
 	} else if runtime.GOOS == "darwin" {
 		d.focusPreviousWindow()

